src/python/zquantum/core/cost_function.py

```diff
@@ -97,7 +93,11 @@ class AnsatzBasedCostFunction(CostFunction):
         estimator: (zquantum.core.interfaces.estimator.Estimator) = estimator used to compute expectation value of target operator 
         gradient_type (str): parameter indicating which type of gradient should be used.
         save_evaluation_history (bool): flag indicating whether we want to store the history of all the evaluations.
-        epsilon (float): epsilon used for calculating gradient using finite difference method.
+        finite_diff_step_size(float): the step size used in finite difference approximation.
+        n_samples (int): number of samples (i.e. measurements) to be used in the estimator. 
+        epsilon (float): an additive/multiplicative error term. The cost function should be computed to within this error term. 
```

Comment:
Shouldn't these be inputs to the Estimator, not Cost Function?

Reply:
They are inputs to the `get_estimated_expectation_values` method in the `Estimator`. The cost function ultimately passes them to the `Estimator`. 

In the context of an optimization loop, the optimizer may adjust error terms (e.g., `epsilon`) after some iterations of the loop and pass the updated error terms to the cost function. That's why the cost function gets these parameters as well.

Reply:
Makes sense – my gut feeling says there's a better way to do that, but I don't have any better alternative, so let it be ;)